index.js

```diff
@@ -1,3 +1,45 @@
-require.addon = require('require-addon')
+const binding = require('./binding')
+const { isNode } = require('which-runtime')
 
-module.exports = require.addon('.', __filename)
+module.exports = binding
+
+module.exports.sodium_malloc = size => {
+  const buf = Buffer.from(binding._sodium_malloc(size))
+  buf.secure = true
+  return buf
+}
+
+module.exports.crypto_generichash = function (output, input, key = undefined) {
```

Comment:
same as below, if we are wrapping anyway, might as well have two private functions?

Reply:
varargs conflict with fastcalls, so using a wrapper was the quickest way i could think of to funnel all calls into fastpath:  https://github.com/holepunchto/sodium-native/blob/82d8a2c12e5336b23c306442d921df2c7279dcb6/binding.c#L3845-L3847

Reply:
ya per above, you should just define two native functions that are explicit (ie never any varargs in c). agree to punt to 5.0.1 tho